string-random-0.1.3.0 testsuite failing to build

For Stackage Nightly:

```
    Building test suite 'string-random-quickcheck' for string-random-0.1.3.0..
    [1 of 2] Compiling Main
    
    /var/stackage/work/unpack-dir/unpacked/string-random-0.1.3.0-31e80448992dc8642295172aee17de0179ff037c15b4eee4b4050a925430c7bf/te
st/string-random-quickcheck.hs:15:40: error:
        • No instance for (random-1.1:System.Random.RandomGen
                             splitmix-0.1.0.3:System.Random.SplitMix.SMGen)
            arising from a use of ‘StringRandom.stringRandom’
        • In the expression: StringRandom.stringRandom g regregexp
          In the second argument of ‘($)’, namely
            ‘\ (QC.QCGen g) _ -> StringRandom.stringRandom g regregexp’
          In the expression:
            QC.MkGen
              $ \ (QC.QCGen g) _ -> StringRandom.stringRandom g regregexp
       |
    15 | regexp = QC.MkGen $ \(QC.QCGen g) _ -> StringRandom.stringRandom g regregexp
       |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/core/test/string-random-quickcheck.hs
+++ b/core/test/string-random-quickcheck.hs
@@ -12,7 +12,7 @@ import qualified Test.Tasty.QuickCheck as TastyQC
 import Text.Regex.PCRE.Heavy (compileM, (=~))
 
 regexp :: QC.Gen Text.Text
-regexp = QC.MkGen $ \(QC.QCGen g) _ -> StringRandom.stringRandom g regregexp
+regexp = QC.MkGen $ \g _ -> StringRandom.stringRandom g regregexp
   where regregexp = "((\\w|" <> classes <> "|\\(\\w+\\))(\\{[0-4],[5-9]\\}|\\*|\\+|))*"
         classes = "\\[^?(\\w|" <> escaped <> "|a-z)+-?\\]"
         escaped = "\\\\[dDwWsSnr\\[\\]|.(){}^$-]"

--- a/quickcheck/src/Test/QuickCheck/StringRandom.hs
+++ b/quickcheck/src/Test/QuickCheck/StringRandom.hs
@@ -42,4 +42,4 @@ import qualified Text.StringRandom as StringRandom
 -- | The 'matchRegexp pat' defines a generator that produces a text that
 -- | matches a regular expression 'pat'
 matchRegexp :: Text.Text -> QC.Gen Text.Text
-matchRegexp txt = QC.MkGen $ \(QC.QCGen g) _ -> StringRandom.stringRandom g txt
+matchRegexp txt = QC.MkGen $ \g _ -> StringRandom.stringRandom g txt

